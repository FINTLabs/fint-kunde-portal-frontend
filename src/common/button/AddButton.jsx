import PropTypes from "prop-types";
import React, { Component } from "react";
import { Tooltip, IconButton } from '@mui/material';
import { withStyles } from '@mui/styles';

import AddIcon from "@mui/icons-material/CheckBoxOutlineBlank";

const styles = theme => ({
  addIcon: {
    //color: theme.palette.secondary.main
  }
});
class AddButton extends Component {
  render() {
    const { classes } = this.props;
    return (
      <Tooltip placement={this.props.placement} title={this.props.title}>
        <IconButton aria-label="Add" onClick={this.props.onClick}>
          <AddIcon className={classes.addIcon} />
        </IconButton>
      </Tooltip>
    );
  }
}

AddButton.defaultProps = {
  placement: "top"
};

AddButton.propTypes = {
  classes: PropTypes.any.isRequired,
  onClick: PropTypes.func,
  placement: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired
};
export default withStyles(styles)(AddButton);
