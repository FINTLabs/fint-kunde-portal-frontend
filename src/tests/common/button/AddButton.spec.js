import React from "react";
import { create } from "react-test-renderer";
import AddButton from "../../../common/button/AddButton";
describe("Feature component", () => {
  test("it matches the snapshot", () => {
    const component = create(<AddButton title="test"/>);
    expect(component.toJSON()).toMatchSnapshot();
  });
});