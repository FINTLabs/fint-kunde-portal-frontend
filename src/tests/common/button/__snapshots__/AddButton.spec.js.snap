// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feature component it matches the snapshot 1`] = `
<button
  aria-describedby={null}
  aria-label="Add"
  className="MuiButtonBase-root MuiIconButton-root"
  disabled={false}
  onBlur={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  title="test"
  type="button"
>
  <span
    className="MuiIconButton-label"
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root AddButton-addIcon-1"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
      />
    </svg>
  </span>
  <span
    className="MuiTouchRipple-root"
  />
</button>
`;
